### ISSUE TITLE
Ampersand in Markdown image URL is not escaped in XML output

### ISSUE THREAD

**Issue by j-ulrich** [doxygen/doxygen#6744]
When a Markdown file contains an image URL with an ampersand, the generated XML output (`md_*.xml`) is invalid because the ampersand is not escaped inside the `name` attribute of the `image` tag inside the XML.

## Example ##
The following Markdown file (`image.md`)
```markdown
![An image](https://img.shields.io/badge/docs-Doxygen-blue.svg?foo&bar)
[A link](http://www.doxygen.nl?foo&bar)
```
leads to this XML (`md_image.xml`):
```xml
<?xml version='1.0' encoding='UTF-8' standalone='no'?>
<doxygen xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="compound.xsd" version="1.8.15">
  <compounddef id="md_image" kind="page">
    <compoundname>md_image</compoundname>
    <title>image</title>
    <briefdescription>
    </briefdescription>
    <detaileddescription>
<para><image type="html" name="docs-Doxygen-blue.svg?foo&bar" inline="yes">></image>
 <ulink url="http://www.doxygen.nl?foo&amp;bar">Some normal link</ulink> </para>
    </detaileddescription>
  </compounddef>
</doxygen>
```
Note that the `name` attribute of the `image` tag contains an unescaped `&` while the `url` attribute of the `ulink` tag contains an escaped `&amp;`.

Aside: `image` tags always contain an literal greater than symbol (`>`). Is this fixed with #6039?

Also note that in the other XML file (`image_8md.xml`), it is escaped properly:
```xml
<?xml version='1.0' encoding='UTF-8' standalone='no'?>
<doxygen xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="compound.xsd" version="1.8.15">
  <compounddef id="image_8md" kind="file" language="Markdown">
    <compoundname>image.md</compoundname>
    <briefdescription>
    </briefdescription>
    <detaileddescription>
    </detaileddescription>
    <programlisting>
<codeline><highlight class="normal">![Some<sp/>SVG<sp/>image](https://img.shields.io/badge/docs-Doxygen-blue.svg?foo&amp;bar)</highlight></codeline>
<codeline><highlight class="normal">[Some<sp/>normal<sp/>link](http://www.doxygen.nl?foo&amp;bar)</highlight></codeline>
    </programlisting>
    <location file="image.md"/>
  </compounddef>
</doxygen>
```

**Comment by j-ulrich** [comment]
As a workaround, one can escape the ampersand manually in the Markdown:
```markdown
![An image](https://img.shields.io/badge/docs-Doxygen-blue.svg?foo&amp;bar)
[A link](http://www.doxygen.nl?foo&bar)
```
This still produces valid links in the HTML.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #6746

**Comment by albert-github** [comment]
Code has been integrated in master on github

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
